### PR TITLE
Orig address and content type added to headers which are passed to the mapper

### PR DESCRIPTION
**Pull request by BobClaerhout**
Signed-off-by: Bob Claerhout <claerhout.bob@gmail.com>

### CODE REVIEW

File: adapters/mqtt-vertx-base/src/main/java/org/eclipse/hono/adapter/mqtt/impl/HttpBasedMessageMapping.java [hunk 1]
```diff
@@ -121,6 +121,8 @@ private void mapMessageRequest(
                 headers.add(property.getKey(), Json.encode(value));
             }
         });
+        headers.add(MessageHelper.APP_PROPERTY_ORIG_ADDRESS, ctx.message().topicName());
+        headers.add(MessageHelper.SYS_PROPERTY_CONTENT_TYPE, ctx.contentType());
```
Comment (sophokles73) [inline adapters/mqtt-vertx-base/src/main/java/org/eclipse/hono/adapter/mqtt/impl/HttpBasedMessageMapping.java:125]: Why don't we use the standard HTTP *Content-Type* header?

Comment (BobClaerhout) [reply adapters/mqtt-vertx-base/src/main/java/org/eclipse/hono/adapter/mqtt/impl/HttpBasedMessageMapping.java]: you're right, the standard HTTP header is better, updated.

**Comment by sophokles73** [comment]
@BobClaerhout not sure if you noticed but the unit tests are failing for your changes ...

**Comment by BobClaerhout** [comment]
> @BobClaerhout not sure if you noticed but the unit tests are failing for your changes ...

Fixed it, test now fails due to something with mongo?

**Comment by sophokles73** [comment]
@BobClaerhout there doesn't seem to be any assertions in the unit tests that verify the inclusion of the new properties.
I also think that it would be good to amend the documentation of the mapping feature in the MQTT adapter guide with the properties that are being forwarded to the mapper so that implementers know what to expect. WDYT?

**Comment by BobClaerhout** [comment]
> @BobClaerhout there doesn't seem to be any assertions in the unit tests that verify the inclusion of the new properties.
> I also think that it would be good to amend the documentation of the mapping feature in the MQTT adapter guide with the properties that are being forwarded to the mapper so that implementers know what to expect. WDYT?

I updated and added some basic documentation as well. Don't know whether this is enough.

**Review by sophokles73 — APPROVED** [review]
LGTM

**Pull request merged**